frontend/lib/resolution/disambiguation.cpp

```diff
@@ -1028,7 +1028,40 @@ static void discardWorseArgs(const DisambiguationContext& dctx,
 static void discardWorseWhereClauses(const DisambiguationContext& dctx,
                                      const CandidatesVec& candidates,
                                      std::vector<bool>& discarded) {
-  // TODO: fill me in
+  int nWhere = 0;
+  int nNoWhere = 0;
+  for (size_t i = 0; i < candidates.size(); ++i) {
+    if (discarded[i]) {
+      continue;
+    }
+
+    const DisambiguationCandidate* candidate = candidates[i];
+    bool where = candidate->fn->whereClauseResult() != TypedFnSignature::WHERE_NONE;
+                // TODO: is there a dyno equivalent of this flag?
+                //  !candidate->fn->hasFlag(FLAG_COMPILER_ADDED_WHERE);
```

Comment:
No, there isn't; this flag is arguably caused by normalizing before resolving, but the dyno resolver does not do that.